src/spatial_ref/srs.rs

```diff
@@ -7,6 +7,148 @@ use std::str::FromStr;
 
 use crate::errors::*;
 
+/// Options for [`CoordTransform::new_ex`].
+#[derive(Debug)]
+pub struct CoordTransformOptions {
+    inner: gdal_sys::OGRCoordinateTransformationOptionsH,
+}
+
+impl Drop for CoordTransformOptions {
+    fn drop(&mut self) {
+        unsafe { gdal_sys::OCTDestroyCoordinateTransformationOptions(self.inner) };
+        self.inner = ptr::null_mut();
+    }
+}
+
+impl CoordTransformOptions {
+    /// Creation options for [`CoordTransform`].
+    pub fn new() -> Result<CoordTransformOptions> {
+        let c_obj = unsafe { gdal_sys::OCTNewCoordinateTransformationOptions() };
+        if c_obj.is_null() {
+            return Err(_last_null_pointer_err(
+                "OCTNewCoordinateTransformationOptions",
+            ));
+        }
+        Ok(CoordTransformOptions { inner: c_obj })
+    }
+
+    /// Sets an area of interest.
+    ///
+    /// The west longitude is generally lower than the east longitude, except for areas of interest
+    /// that go across the anti-meridian.
+    ///
+    /// # Arguments
+    ///
+    /// - `west_longitude_deg` – West longitude (in degree). Must be in [-180,180]
+    /// - `south_latitude_deg` – South latitude (in degree). Must be in [-90,90]
+    /// - `east_longitude_deg` – East longitude (in degree). Must be in [-180,180]
+    /// - `north_latitude_deg` – North latitude (in degree). Must be in [-90,90]
+    pub fn set_area_of_interest(
+        &mut self,
+        west_longitude_deg: f64,
+        south_latitude_deg: f64,
+        east_longitude_deg: f64,
+        north_latitude_deg: f64,
+    ) -> Result<()> {
+        let ret_val = unsafe {
+            gdal_sys::OCTCoordinateTransformationOptionsSetAreaOfInterest(
+                self.inner,
+                west_longitude_deg,
+                south_latitude_deg,
+                east_longitude_deg,
+                north_latitude_deg,
+            )
+        };
+        if ret_val == 0 {
+            return Err(GdalError::OctError {
+                method_name: "OCTCoordinateTransformationOptionsSetAreaOfInterest",
+            });
+        }
+        Ok(())
+    }
+
+    /// Sets the desired accuracy for coordinate operations.
+    ///
+    /// Only coordinate operations that offer an accuracy of at least the one specified will be
+    /// considered.
+    ///
+    /// An accuracy of 0 is valid and means a coordinate operation made only of one or several
+    /// conversions (map projections, unit conversion, etc.) Operations involving ballpark
+    /// transformations have a unknown accuracy, and will be filtered out by any dfAccuracy >= 0
+    /// value.
+    ///
+    /// If this option is specified with PROJ < 8, the `OGR_CT_OP_SELECTION` configuration option
+    /// will default to `BEST_ACCURACY`.
+    #[cfg(all(major_ge_3, minor_ge_3))]
```

Comment:
This is wrong, it will fail in `4.0`. We want `major > 3 || major == 3 && minor >= 3`. There are some places where we do this correctly.

But honestly, I wish we could improve on these somehow.

Reply:
Done. Yeah, definitely not the only occurrence of this mistake in the codebase, or even in this file.

Merging major and minor into a single `gdal_ge_3_3` option would make these lines shorter, at the expense of adding a bunch of (autogenerated) cfg options.

A custom proc macro with custom syntax could of course also work.